Keas.Mvc/Controllers/TeamAdminController.cs

```diff
@@ -746,20 +752,340 @@ public async Task<IActionResult> Upload(IFormFile file)
+                                    if (localPerson != null) // I'm using a extension method
+                                    {
+                                        // detach
+                                        _context.Entry(localPerson).State = EntityState.Detached;
+                                    }
+                                }
+                            }
+                        }
+
+                        if (!result.Success)
+                        {
+                            result.Messages = new List<string>();
+                        }
+                        resultsView.Add(result);
+                    }
+                }
+            }
+
+            Message = $"Successfully loaded {equipmentCount} new items, {peopleCount} new or reactivated team members, and {assignmentCount} new assignments recorded.";
+            if (errorCount > 0)
+            {
+                ErrorMessage = $"{errorCount} rows not imported due to errors.";
+            }
+            return View(resultsView);
+
+        }
+
         private static string SetStatus(string status, KeyImportResults result)
         {
             switch (status)
             {
                 case "Active":
-                    return("Active");
+                    return ("Active");
                 case "Lost":
-                    return("Lost");
+                    return ("Lost");
                 case "Destroyed":
-                    return("Destroyed");
+                    return ("Destroyed");
                 default:
                     result.Messages.Add("Key status defaulted to Active.");
-                    return("Active");
+                    return ("Active");
+            }
+        }
+
+        private void CreateAttribute(Equipment equipment, string key, string value, EquipmentImportResults result, ref int recAttributeCount, ref bool recAttributeAdded)
+        {
+            var equipmentAttribute = new EquipmentAttribute();
+            equipmentAttribute.Equipment = equipment;
+            equipmentAttribute.Key = key;
+            equipmentAttribute.Value = value;
+            ModelState.Clear();
+            TryValidateModel(equipmentAttribute);
+            if (ModelState.IsValid)
+            {
+                _context.EquipmentAttributes.Add(equipmentAttribute);
+                recAttributeCount += 1;
+                recAttributeAdded = true;
+            }
+            else
+            {
+                result.Success = false;
+                result.ErrorMessage.Add($"Invalid Equipment Attribute values Error(s): {GetModelErrors(ModelState)} ");
+            }
+        }
+
+        private Equipment CreateEquipment(EquipmentImport r, Team team, EquipmentImportResults result, ref int recEquipmentCount)
+        {
+            var equipment = new Equipment();
+            if (!string.IsNullOrWhiteSpace(r.EquipmentName) && !r.Make.Contains("AEXMPLE"))
```

Comment:
@jcubbage I'd say remove this if. We don't really need it. If they import junk they can delete it.

Reply:
We left it in the Key import.

Reply:
@jcubbage Keys are a little more difficult to remove. And it probably shouldn't be in there either.

Reply:
Pulled from both. Updated message.